### PR TITLE
Avoid conflict when run add test

### PR DESCRIPTION
**Pull request by sivchari**
## Description

To avoid conflict, I added new template to generate test and separate marker tests for each file.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Refactoring (no functional changes)

## Quality Checklist

Before submitting this pull request, ensure your contribution meets these quality standards:

### Code Quality
- [x] Code passes `make golangci-lint` without errors
- [x] All tests pass: `make test`
- [x] Code follows existing patterns and conventions
- [x] No hard-coded values (use constants or configuration)
- [x] Error messages are clear and actionable

### Testing Requirements
- [ ] Golden tests created and passing (`internal/analyzers/govalid/testdata/`)
- [ ] Unit tests implemented with boundary value testing (`test/unit/`)
- [ ] Benchmark tests comparing against popular validation libraries (`test/benchmark/`)
  - [ ] [go-playground/validator](https://github.com/go-playground/validator) (BenchmarkGoPlayground*)
  - [ ] [asaskevich/govalidator](https://github.com/asaskevich/govalidator) (BenchmarkGoValidator*)
  - [ ] [go-ozzo/ozzo-validation](https://github.com/go-ozzo/ozzo-validation) (BenchmarkOzzoValidation*)
  - [ ] [gookit/validate](https://github.com/gookit/validate) (BenchmarkGookitValidate*)
- [ ] All test files generated: `cd test && go generate`

### Performance Standards
- [ ] Zero allocations in validation logic (verified via benchmarks)
- [ ] Performance improvement over existing validation libraries (minimum 2x faster)
- [ ] Benchmark results added to `test/benchmark/README.md`

### Documentation
- [ ] README.md updated with new marker documentation
- [ ] Code includes appropriate comments and examples
- [ ] Validator usage examples provided

### Integration
- [ ] Validator scaffold generated with `make generate-validator MARKER=yourmarker`
- [ ] Registry files automatically updated
- [ ] Binary rebuilt and installed: `go install ./cmd/govalid/`

### Final Verification
- [ ] All GitHub Actions CI checks pass
- [ ] No breaking changes to existing functionality
- [ ] Contribution follows project license requirements

### CODE REVIEW

**Comment by sivchari** [comment]
@taua-almeida @BlackBuck @ccoVeille 
If you have time, please review it.

**Comment by sivchari** [comment]
After #111 is merged, I'll merge it 👍

File: cmd/generate-validators/internal/generate/generator.go [hunk 1]
```diff
@@ -43,6 +43,21 @@ func All(rulesDir, outputDir, registryFile, markersFile string, templates Templa
 		return fmt.Errorf("failed to generate markers file: %w", err)
 	}
 
+	// Generate test files
+	testDir := "internal/analyzers/govalid/tests"
```
Comment (ccoVeille) [inline cmd/generate-validators/internal/generate/generator.go:47]: I think that if you want the test to pass also on Windows, you have to use this


```suggestion
	testDir := filepath.Join("internal","analyzers","govalid","tests")
```

File: cmd/generate-validators/internal/scaffold/validator.go [hunk 2]
```diff
@@ -60,3 +60,39 @@ func checkFileExists(path string) error {
...
+	// Convert marker name to various forms
+	markerLower := strings.ToLower(markerName)
+
+	// Simple inline PascalCase conversion
+	r, size := utf8.DecodeRuneInString(markerLower)
+	if size == 0 {
+		return errors.New("marker name cannot be empty")
+	}
+
+	titleCaseName := strings.ToUpper(string(r)) + markerLower[size:]
```
Comment (ccoVeille) [inline cmd/generate-validators/internal/scaffold/validator.go:79]: Why not using toTitle function ?

File: cmd/generate-validators/main.go [hunk 3]
```diff
@@ -31,12 +31,16 @@ var markersTemplate string
...
 const (
-	rulesDir     = "internal/validator/rules"
-	outputDir    = "internal/validator/registry/initializers"
-	registryFile = "internal/analyzers/govalid/registry_init.go"
-	markersFile  = "internal/markers/markers_generated.go"
+	rulesDir       = "internal/validator/rules"
+	outputDir      = "internal/validator/registry/initializers"
+	registryFile   = "internal/analyzers/govalid/registry_init.go"
+	markersFile    = "internal/markers/markers_generated.go"
+	govalidTestDir = "internal/analyzers/govalid/tests"
```
Comment (ccoVeille) [inline cmd/generate-validators/main.go:43]: Sane here about using filepath.Join
